### ISSUE TITLE
Can't reproduce the reported results

### ISSUE THREAD

**Issue by trahman888** [XgDuan/WSDEC#7]
Hi,
Thanks for providing code in github. I am able to run your code. But unfortunately could not able to generate you reported results. Even I tried using your pretrained model from Onedrive specially for CIDEr, the scores are far away from the reported results. I followed your approach described in readme. Can you please tell me how can  I able to reproduce the results?

**Comment by XgDuan** [comment]
What's your training script and the result (and the training epochs) using our provided pretrained model? I would like to find out the problem with you.

**Comment by trahman888** [comment]
I used training script that you provided. For pretrained caption generation I got 7.26 CIDEr score. And the number of epoch is 100 and I tested the result after every 10 epoch.

**Comment by XgDuan** [comment]
@trahman888 
Thanks for your feedback. I think there may be something different when creating this Github public repo. I cloned the code on this repo and rerun the experiment. Here are the suggestions.

1. The pre-training part. You should not use the model at the 100th epoch. The best result is achieved at about 20th epochs. As I have frozen the random seed, you can compare the results between yours and ours: our result (re-runed today) is 9.12 CIDEr at 20th epoch, 9.22 at 21st epoch, and the result drops to 8.88 at 41st epoch (I do not have more time to run 100 epochs, but I think you can check your result easily, and our provided pre-trained model achieves about 8.8 CIDEr score. ). **The suggestion is that: 1) set the `test_interval = 1` to track every result,  2) check those results at early epochs instead of the 100th (overfitting).**

2. To train the final model, the aforementioned suggestions are also useful. I am running the new experiment to make sure that everything is fine, and I will update later.  UPDATE: the result we obtained by running the Github code is consistent with our original result. But it's worth to mention that the best result is obtained in the first several epochs (4th in our case) as that the model is not a complex one (several FC layers and an LSTM layer) and the dataset contains abundant data. We will change the initial `max-epoch` parameter to ease this misleading.

**Comment by XgDuan** [comment]
@trahman888 Are you still there?
It seems that there is something wrong with our code instruction or your evaluation process (We received the email from your mentor). So I would like to know about your current progress and problem. Could you please tell me about what have you done currently and what's the current scores?

**Comment by trahman888** [comment]
Thanks for your cooperation. Actually I am able to reproduce results for other metric except CIDER score. I run evaluate_pretrain.py for evaluating pretrain model and  evaluate.py for final model. I also set test_interval=1. As cider score, I got 8.63 and 15.19  for pretrain and final model respectively.

**Comment by XgDuan** [comment]
@trahman888 

Well, to better understand your situation and figure out the problem, I do have several more questions: 1) when do you obtain the 15.19 cider score? And would you mind sending your final evaluation result to me(email: lazydxg@gmail.com) 2) have you ever tried using the pretrained model we provided? If not, would you please try to train a final model using the cg_pretrain.pkl in [final.zip](https://onedrive.live.com/?authkey=%21AKA59rVk4%2DS0tNc&cid=F54CB0508940633C&id=F54CB0508940633C%214507&parId=F54CB0508940633C%214487&action=defaultclick). 

If using our provided final model reproduces the final result, I think the problem is due to the pretraining process, my suggestion is choosing another pretrain model(Our pretraining model is selected from several candidates, instead of the BEST one. And I tested the final training processing (use the above mentioned pretrained model) when preparing this GitHub repo, but the pretraining process maybe a little different I think?)

Please let me know when you have any progress or more problems.

**Comment by XgDuan** [comment]
One more thing. When choosing a pretrained-model, we do not just find the one that has the best CIDER score. Instead, we will try to find a balanced one who performs well on all metrics.

**Comment by trahman888** [comment]
(1) I got the score after 12 epoch. I have sent my final evaluation results to you. (2) Sure, I can give a try to train the final model using the pretrain model you provided and let you know.

Thanks for your help.

**Comment by XgDuan** [comment]
This issue will be closed for that we have help @trahman888 solve their problem.

One suggestion is that:
When training the final model, we should `try' several different pretraining models and select the best one instead of focusing on a certain selected one.
This is a little tricky but indeed works. We would also consider improving the robustness or smoothing the gap between pretraining and final-training as future work.
